GameTracker-Agent/ViewModels/MainWindowViewModel.cs

```diff
@@ -128,63 +121,56 @@ public void AddDirectory(object obj)
                 EnsureValidNames = true,
                 Multiselect = false,
                 ShowPlacesList = true
-            };
-
-            if (dlg.ShowDialog() == CommonFileDialogResult.Ok)
+            })
             {
-                controller.addGameDirectory(dlg.FileName);
-                var gameDirectory = controller.GetGameDirectory(dlg.FileName);
-                GameDirectories.Add(new GameDirectoryDto(gameDirectory.Directory, gameDirectory.GetGames()));
-                //RaisePropertyChanged();
-                //MessageBox.Show(gameDirectory.Directory);
+                if (dlg.ShowDialog() == CommonFileDialogResult.Ok)
+                {
+                    controller.AddGameDirectory(dlg.FileName);
+                    var gameDirectory = controller.GetGameDirectory(dlg.FileName);
+                    var gameDirectoryDto = new GameDirectoryDto(gameDirectory.Directory, gameDirectory.GetGames());
+                    GameDirectories.Add(gameDirectoryDto);
+                    selectedDirectory = GameDirectories.Last();
```

Comment:
Set the property not the member (change propagation)
Also why `Last` here and `First` in the c'tor?